components/filemanager/toolbar.md

```diff
@@ -10,36 +10,403 @@ position: 15
+| Sort Direction | A ButtonGroup with ToggleButtons that allows selecting the sort direction(ascending or descending) to sort the files in the current location. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Sort By | A SplitButton that provides an option to select the desired sort member to sort by. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article. |
+| Views | A ButtonGroup with ToggleButtons that allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager - Views]({%slug filemanager-views%}) article. |
+| View Details | A Switch that toggles the visibility of the [Preview Pane]({%slug filemanager-preview-pane%}). By default its value is `Off` and you can enable it to see the selected file details. |
+| Search | A TextBox that allows you to filter the files by name.  Read more in the [FileManager - Search]({%slug filemanager-search%}) article. |
 
-FileManager allows sorting of the files in the current location. The sorting direction (ascending or descending) is controlled through dedicated buttons in the Toolbar. The `SortBy` dropdown provides option to select the desired sort member to sort by. Read more in the [FileManager - Sort]({%slug filemanager-sort%}) article.
+## Custom Toolbar
 
+To customize the Toolbar, add the `<FileManagerToolBar>` tag as a direct child of the `<TelerikFileManager>`. Inside the `<FileManagerToolBar>` you can include your desired [built-in tools](#built-in-tools), manage their order and add [custom tools](#custom-tools).
 
-## Views
+### Built-in tools
 
-The `Views` button group allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager - Views]({%slug filemanager-views%}) article.
+To add the desired built-in tools, declare the corresponding tool components in the `<FileManagerToolBar>`. The tools will be rendered in the order that you declare them in.
 
-## View Details
+<div class="skip-repl"></div>
 
-The `View Details` toggles the visibility of the [Preview Pane]({%slug filemanager-preview-pane%}). By default is is `Off` and you can enable it to see the selected file details.
+````CSHTML
+<TelerikFileManager Data="@FileManagerData">
+                   <FileManagerToolBar>
+                       <FileManagerToolBarSearchTool/>
+                   </FileManagerToolBar>
+</TelerikFileManager>
+````
 
-## Search
+>caption Built-in tool components:
 
-The FileManager allows searching files in the current location via a Searchbar in the Toolbar. Read more in the [FileManager - Search]({%slug filemanager-search%}) article.
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Tool Component | Description |
```

Comment:
Optionally, you can consider a single table for the built-in tools and tool tags. This will save space and avoid repetition:
https://docs.telerik.com/blazor-ui/components/pdfviewer/toolbar